### PR TITLE
feat: add gitpod support

### PR DESCRIPTION
**Pull request by mihai-sysbio**
This PR makes it possible to run the entire stack in Gitpod, and adds a label in the Readme for it. @mblue9 and I started to look into this during the hackathon but didn’t quite get it to work. It proved to be a restriction on the allowed hosts. 

image

### CODE REVIEW

**Comment by mblue9** [comment]
Great! 
I had looked into this and found similar but didn't get around to adding it yet :)

Not part of this PR, but the other related things we discussed in case I forget, 

- if we want to make the preview easier, it could automatically appear with `open-preview` in .gitpod.yml e.g. see what Galaxy do here https://github.com/galaxyproject/training-material/blob/main/.gitpod.yml
- also note as in their yml, you can automate initial setup tasks e.g would it make sense to automate the creation of .env (`cp .env.default .env`) and `cp postgres/postgres.env.default postgres/postgres.env`. I didn't initially know to do those steps as they're not something I've done before, maybe others are the same.

**Review by mblue9 — APPROVED** [review]

**Comment by mihai-sysbio** [comment]
Those were too good suggestions to wait with them @mblue9 so I've added them though commit [dba8fa3](https://github.com/elixir-europe-training/ELIXIR-TrP-Training-Metrics-Database-Tango/pull/28/commits/dba8fa331a7c4ce5136be1da536a4c7c7dd7bd43).

**Review by jonandernovella — CHANGES_REQUESTED** [review]
Allowing all hosts by default is not a good idea.

**Comment by mihai-sysbio** [comment]
> Allowing all hosts by default is not a good idea.

@jonandernovella could you expand?

Here's my thinking: For development, the Dockerfile is now set up to only allow requests from the localhost. Imho, this should be more secure than the filter in Django. And for production, this filter should not matter. Also, I think accepting all hosts is _required_ for Django to work in Gitpod - but perhaps not?

**Comment by jonandernovella** [comment]
> > Allowing all hosts by default is not a good idea.
> 
> @jonandernovella could you expand?
> 
> Here's my thinking: For development, the Dockerfile is now set up to only allow requests from the localhost. Imho, this should be more secure than the filter in Django. And for production, this filter should not matter. Also, I think accepting all hosts is _required_ for Django to work in Gitpod - but perhaps not?

If gitpod requires accepting all hosts that's OK. But we shouldn't need to change the default env values, which are secure by default. Instead I'd prefer if you can customize the actual env file to meet the requirements for gitpod.

**Comment by mihai-sysbio** [comment]
> But we shouldn't need to change the default env values, which are secure by default.

In my view, this Django-provided security is of little/no benefit over the Docker-provided security, and the project is now fully running inside Docker.

> Instead I'd prefer if you can customize the actual env file to meet the requirements for gitpod.

Good idea, that is of course possible - another env/Dockerfile can be set up especially for Gitpod, and to me that is extra maintenance (and possible duplication) which is why I prefer to avoid it.

However, @jonandernovella if you feel strongly about these changes, please feel free to add them directly to this branch.

**Comment by jonandernovella** [comment]
> > But we shouldn't need to change the default env values, which are secure by default.
> 
> In my view, this Django-provided security is of little/no benefit over the Docker-provided security, and the project is now fully running inside Docker.
> 
> > Instead I'd prefer if you can customize the actual env file to meet the requirements for gitpod.
> 
> Good idea, that is of course possible - another env/Dockerfile can be set up especially for Gitpod, and to me that is extra maintenance (and possible duplication) which is why I prefer to avoid it.
> 
> However, @jonandernovella if you feel strongly about these changes, please feel free to add them directly to this branch.

I don't think a new Dockerfile is needed. You'd just need to modify that variable in the gitpod script.

File: .env.default [hunk 1]
```diff
@@ -1,4 +1,4 @@
-DJANGO_ALLOWED_HOSTS=localhost
+DJANGO_ALLOWED_HOSTS=*
```
Comment (unode) [inline .env.default:1]:
```suggestion
DJANGO_ALLOWED_HOSTS=localhost,.gitpod.io
```
Also suggesting to keep the restriction and be explicit about other sources.

**Comment by unode** [comment]
Thanks @mihai-sysbio for this setup. I'm also a big fan of GitPod. Massive time saver.

File: .env.default (hunk 1 above)
Comment (mihai-sysbio) [reply .env.default]: I've committed the above without testing.

File: .gitpod.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+ports:
+  - name: Django
+    description: the port used by the Django container
+    port: 8000
+    onOpen: open-preview
+    visibility: public
+  - name: Postgres
+    description: the port used by the Postgres container
+    port: 5432
+    onOpen: ignore
```
[the remark this answers is not in the document]
Comment (mihai-sysbio) [reply .gitpod.yml]: Added in [8d1deb5](https://github.com/elixir-europe-training/ELIXIR-TrP-Training-Metrics-Database-Tango/pull/28/commits/8d1deb5423330dae85bf5385c525aaca9562785d)

File: .env.default (hunk 1 above)
Comment (mihai-sysbio) [reply .env.default]: Tested, it works 👍🏻

**Review by jonandernovella — APPROVED** [review]

**Pull request merged**